packages/core/src/renderer/components/+namespaces/namespace-tree-view.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import styles from "./namespace-tree-view.module.scss";
+
+import { SvgIcon } from "@material-ui/core";
+import { TreeItem, TreeView } from "@material-ui/lab";
+import { withInjectables } from "@ogre-tools/injectable-react";
+import React from "react";
+import { Link } from "react-router-dom";
+import type { Namespace } from "../../../common/k8s-api/endpoints";
+import { DrawerTitle } from "../drawer";
+import type { GetDetailsUrl } from "../kube-detail-params/get-details-url.injectable";
+import getDetailsUrlInjectable from "../kube-detail-params/get-details-url.injectable";
+import { SubnamespaceBadge } from "./subnamespace-badge";
+import hierarchicalNamespacesInjectable from "./hierarchical-namespaces.injectable";
+
+interface NamespaceTreeViewProps {
+  root: Namespace;
+}
+
+interface Dependencies {
+  namespaces: Namespace[];
+  getDetailsUrl: GetDetailsUrl;
+}
+
+function isNamespaceControlledByHNC(namespace: Namespace) {
+  const hierarchicalNamesaceControllerLabel = "hnc.x-k8s.io/included-namespace=true";
+
+  return namespace.getLabels().find(label => label === hierarchicalNamesaceControllerLabel);
+}
+
+function NonInjectableNamespaceTreeView({ root, namespaces, getDetailsUrl }: Dependencies & NamespaceTreeViewProps) {
+  const [expandedItems, setExpandedItems] = React.useState<string[]>(namespaces.map(ns => `namespace-${ns.getId()}`));
+  const classes = { group: styles.group, label: styles.label };
+  const nodeId = `namespace-${root.getId()}`;
+
+  function renderChildren(parent: Namespace) {
+    const children = namespaces.filter(ns =>
+      ns.getLabels().find(label => label === `${parent.getName()}.tree.hnc.x-k8s.io/depth=1`),
```

Comment:
Again I'd better to keep such pieces if api-data as `*.tree.hnc.x-k8s.io/depth=1` closer to namespace api (for easier future maintenance at least)